crates/client/src/lib.rs

```diff
@@ -1222,6 +1225,10 @@ impl FileSystemClient {
         let disable_interactive =
             cfg!(not(feature = "cli-interactive")) || config.disable_interactive;
 
+        if cfg!(feature = "keyring") && auth_token.is_none() && config.keyring_auth {
+            auth_token = crate::keyring::get_auth_token(&url)?
+        }
+
```

Comment:
Does this compile with `cargo build --no-default-features`? We should probably add that as a check for CI...

Reply:
It compiles for me.

Reply:
```
$ cd crates/client && cargo build --no-default-features
...
error[E0433]: failed to resolve: could not find `keyring` in the crate root
```

Reply:
Pushed fixed.